config/tf_modules/aws-cloudfront-distribution/main.tf

```diff
@@ -0,0 +1,92 @@
+resource "random_string" "default-suffix" {
+  length  = 7
+  special = false
+}
+
+data "aws_s3_bucket" "current_bucket" {
+  bucket = var.bucket_name
+}
+
+resource "aws_s3_bucket_object" "default_page" {
+  bucket       = data.aws_s3_bucket.current_bucket.id
+  key          = "opta-default-page-${random_string.default-suffix.result}.html"
+  source       = "${path.module}/default.html"
+  content_type = "text/html"
+}
+
+resource "aws_cloudfront_distribution" "s3_distribution" {
+  origin {
+    domain_name = data.aws_s3_bucket.current_bucket.bucket_regional_domain_name
+    origin_id   = local.s3_origin_id
+
+    s3_origin_config {
+      origin_access_identity = var.origin_access_identity_path
+    }
+  }
+
+  enabled             = true
+  is_ipv6_enabled     = true
+  comment             = "Opta managed cloudfront distribution ${var.layer_name}-${var.module_name}"
+  default_root_object = var.default_page_file == null ? aws_s3_bucket_object.default_page.key : var.default_page_file
+
+  dynamic "logging_config" {
+    for_each = var.s3_log_bucket_name == null ? [] : [1]
+    content {
+      include_cookies = true
+      bucket          = var.s3_log_bucket_name
```

Comment:
I don't think we should re-use the s3 logging bucket in cloudfront.

Reply:
... why not? This is the environment-wide audit logging bucket used to store logs from the load balancers, buckets, and anything else that supports sending audits to an s3 bucket.

Reply:
Oh sorry I misunderstood it. It's coming from the parent.. that's fine then!